When using kotlin with mongo panache, the deserializer fails unless all the constructor parameters are declared as nullable or have a default value

The title is pretty much self explanatory. When using a kotlin class that extends PanacheMongoEntity unless the fields are declared as nullable or have default values the deserialization when doing any find operation (e.g. listAll()) fails with the following error:
```
Caused by: org.bson.codecs.configuration.CodecConfigurationException: Cannot find a public constructor for 'ChargingInterval'.
	at org.bson.codecs.pojo.CreatorExecutable.checkHasAnExecutable(CreatorExecutable.java:140)
	at org.bson.codecs.pojo.CreatorExecutable.getInstance(CreatorExecutable.java:107)
	at org.bson.codecs.pojo.InstanceCreatorImpl.<init>(InstanceCreatorImpl.java:40)
	at org.bson.codecs.pojo.InstanceCreatorFactoryImpl.create(InstanceCreatorFactoryImpl.java:28)
	at org.bson.codecs.pojo.ClassModel.getInstanceCreator(ClassModel.java:71)
	at org.bson.codecs.pojo.PojoCodecImpl.decode(PojoCodecImpl.java:120)
	at org.bson.codecs.pojo.PojoCodecImpl.decode(PojoCodecImpl.java:125)
	at org.bson.codecs.pojo.AutomaticPojoCodec.decode(AutomaticPojoCodec.java:37)
```

Sample failing class:
```
package io.biapower.services.chargingsessionservice.domain
import io.quarkus.mongodb.panache.PanacheMongoEntity
import java.time.Instant

data class SampleDataClass(
        var start: Instant = Instant.now(),
        var end: Instant,
        var num: Int = 0f
): PanacheMongoEntity()
```


Sample working class:
```
package io.biapower.services.chargingsessionservice.domain
import io.quarkus.mongodb.panache.PanacheMongoEntity
import java.time.Instant

data class SampleDataClass(
        var start: Instant = Instant.now(),
        var end: Instant = Instant.now(),
        var num: Int = 0
): PanacheMongoEntity()
```
or:
```
package io.biapower.services.chargingsessionservice.domain
import io.quarkus.mongodb.panache.PanacheMongoEntity
import java.time.Instant

data class SampleDataClass(
        var start: Instant = Instant.now(),
        var end: Instant? = null,
        var num: Int = 0
): PanacheMongoEntity()
```

--- a/docs/src/main/asciidoc/mongodb-panache.adoc
+++ b/docs/src/main/asciidoc/mongodb-panache.adoc
@@ -565,6 +565,21 @@ data class Person @BsonCreator constructor (
 ): PanacheMongoEntity()
 ----
 
+The last option is to the use the link:https://kotlinlang.org/docs/reference/compiler-plugins.html#no-arg-compiler-plugin[no-arg] compiler plugin.
+This plugin is configured with a list of annotations, and the end result is the generation of no-args constructor for each class annotated with them.
+
+For MongoDB with Panache, you could use the `@MongoEntity` annotation on your data class for this:
+
+[source,kotlin]
+----
+@MongoEntity
+data class Person (
+    var name: String,
+    var birth: LocalDate,
+    var status: Status
+): PanacheMongoEntity()
+----
+
 == How and why we simplify MongoDB API
 
 When it comes to writing MongoDB entities, there are a number of annoying things that users have grown used to

